aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -0,0 +1,641 @@
+            dim_order = [dim_order] * num_scenes
+        if image_name is None or isinstance(image_name, str):
+            image_name = [image_name] * num_scenes
+        if isinstance(pixels_physical_size, tuple):
+            pixels_physical_size = [pixels_physical_size] * num_scenes
+        elif pixels_physical_size is None:
+            pixels_physical_size = [(1.0, 1.0, 1.0)] * num_scenes
+        if channel_names is None or isinstance(channel_names[0], int):
+            channel_names = [channel_names] * num_scenes  # type: ignore
+        if channel_colors is None or isinstance(channel_colors[0], int):
+            channel_colors = [channel_colors] * num_scenes  # type: ignore
+
+        xml = ""
+        # try to construct OME from params
+        if ome_xml is None:
+            ome_xml = OmeTiffWriter.build_ome(
+                [i.shape for i in data],
+                [i.dtype for i in data],
+                channel_names=channel_names,  # type: ignore
+                image_name=image_name,
+                pixels_physical_size=pixels_physical_size,
+                channel_colors=channel_colors,  # type: ignore
+                dimension_order=dim_order,
+            )
+        # else if string, then construct OME from string
+        elif isinstance(ome_xml, str):
+            ome_xml = from_xml(ome_xml)
+
+        # if we do not have an OME object now, something is wrong
+        if not isinstance(ome_xml, OME):
+            raise ValueError(
+                "Unknown OME-XML metadata passed in. Use OME object, or xml string or \
+                None"
+            )
+
+        # vaidate ome
+        for scene_index in range(len(data)):
+            OmeTiffWriter._check_ome_dims(
+                ome_xml, scene_index, data[scene_index].shape, data[scene_index].dtype
+            )
+
+        # convert to string for writing
+        xml = to_xml(ome_xml).encode()
+
+        # now the heavy lifting. assemble the raw data and write it
+
+        for scene_index in range(len(data)):
+            # Assumption: if provided a dask array to save, it can fit into memory
+            if isinstance(data[scene_index], da.core.Array):
+                data[scene_index].compute()  # type: ignore
+
+        # Save image to tiff!
+        tif = tifffile.TiffWriter(
+            path,
+            bigtiff=OmeTiffWriter._size_of_ndarray(data=data) > BIGTIFF_BYTE_LIMIT,
+        )
+
+        for scene_index in range(len(data)):
+            description = xml if scene_index == 0 else None
+            # assume if first channel is rgb then all of it is
+            is_rgb = (
+                ome_xml.images[scene_index].pixels.channels[0].samples_per_pixel > 1
+            )
+            photometric = (
+                TIFF.PHOTOMETRIC.RGB if is_rgb else TIFF.PHOTOMETRIC.MINISBLACK
+            )
+            planarconfig = TIFF.PLANARCONFIG.CONTIG if is_rgb else None
+            tif.write(
+                data[scene_index],
```

Comment:
I would do the compute in this for loop.

```python
for scene_index in range(len(data)):
    if isinstance(data[scene_index], da.core.Array):
        image_data = data[scene_index].compute()
    else:
        image_data = data[scene_index]

# ...

tif.write(
    image_data,
   # ...
)
```

This means you will only ever have one scene in memory at a time if you were provided all dask arrays.

Reply:
I take it plane by plane or chunk by chunk writing wouldn't be possible?

Reply:
I am changing the loop to do what you suggested -- I wasn't trusting the size check to happen before compute() but I feel safe doing it now.

As for per-plane or per-chunk writing, I am sure we can do it in some manner through tifffile's api but I'm not ready to write that code yet.